polaris-react/src/components/DropZone/DropZone.scss

```diff
@@ -87,6 +104,12 @@ $dropzone-stacking-order: (
       border-color: var(--p-color-border-strong);
     }
   }
+
+  #{$se23} &:not(.focused, .isDisabled) {
```

Comment:
the additional `.isDisabled` argument in this :not invocation is to ensure we don't mess up disabled styles due to specificity. 

Won't be necessary post SE23.